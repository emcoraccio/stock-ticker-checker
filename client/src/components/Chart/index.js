import React from 'react';
import { Chart } from 'react-google-charts';


const stockChart = ({ chartData: { symbol, price, pastStats } }) => {

  const data = [
    ["Time", "Price"],
  ];

  pastStats.forEach(stock => {
    let stockData = [stock.time, stock.price];

    data.push(stockData);
  });


  const options = {
    hAxis: {
      title: 'Time',
    },
    vAxis: {
      title: 'Price',
    },
    title: `Price of ${symbol} Over Time`,
    legend: { position: "bottom" },

  };


  return (
    <Chart
      chartType="LineChart"
      data={data}
      options={options}
      width="80%"
      height="200px"
    />
  );

}

export default stockChart;