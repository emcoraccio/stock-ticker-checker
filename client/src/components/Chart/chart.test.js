import React from 'react';
import { shallow } from 'enzyme';
import Chart from './index';

it('expects to render the Chart component', () => {
  const mockChartData = {
    symbol: "A",
    price: "12",
    pastStats: []
  }
  const mockChartData2 = {
    symbol: "A",
    price: "12",
    pastStats: [
      {
        price: 10,
        time: 1582318797546
      },
      {
        price: 11.24,
        time: 1582318797610
      }
    ]
  }
  expect(shallow(<Chart chartData={mockChartData} />)).toMatchSnapshot();
  expect(shallow(<Chart chartData={mockChartData2} />)).toMatchSnapshot();
});