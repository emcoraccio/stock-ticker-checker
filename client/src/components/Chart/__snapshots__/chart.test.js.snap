// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expects to render the Chart component 1`] = `ShallowWrapper {}`;

exports[`expects to render the Chart component 2`] = `ShallowWrapper {}`;
